src/sentry/integrations/msteams/utils.py

```diff
@@ -54,3 +55,376 @@ def get_channel_id(organization, integration_id, name):
+        "type": "Input.ChoiceSet",
+        "value": "0",
+        "id": "resolveInput",
+        "isVisible": False,
+        "choices": [
+            {"title": "Immediately", "value": "0"},
+            {"title": "In the current release", "value": "1"},
+            {"title": "In the next release", "value": "2"},
+        ],
+    }
+
+    submit_card = {
+        "type": "ActionSet",
+        "id": "resolveSubmit",
+        "isVisible": False,
+        "actions": [
+            {"type": "Action.Submit", "title": "Resolve", "data": {"actionType": "resolve"}}
+        ],
+    }
+
+    return [title_card, input_card, submit_card]
+
+
+def build_incident_ignore_card():
+    title_card = {
+        "type": "TextBlock",
+        "size": "Large",
+        "text": "Ignore until this happens again...",
+        "weight": "Bolder",
+        "id": "ignoreTitle",
+        "isVisible": False,
+    }
+
+    input_card = {
+        "type": "Input.ChoiceSet",
+        "value": "0",
+        "id": "ignoreInput",
+        "isVisible": False,
+        "choices": [
+            {"title": "1 time", "value": "0"},
+            {"title": "10 times", "value": "1"},
+            {"title": "100 times", "value": "2"},
+            {"title": "1,000 times", "value": "3"},
+            {"title": "10,000 times", "value": "4"},
+        ],
+    }
+
+    submit_card = {
+        "type": "ActionSet",
+        "id": "ignoreSubmit",
+        "isVisible": False,
+        "actions": [{"type": "Action.Submit", "title": "Ignore", "data": {"actionType": "ignore"}}],
+    }
+
+    return [title_card, input_card, submit_card]
+
+
+def build_incident_assign_card(group):
+    teams = [
+        {"title": u"#{}".format(u.slug), "value": u"team:{}".format(u.id)}
+        for u in group.project.teams.all()
+    ]
+    teams.sort()
+    teams = [{"title": "Me", "value": "0"}] + teams
+    title_card = {
+        "type": "TextBlock",
+        "size": "Large",
+        "text": "Assign to...",
+        "weight": "Bolder",
+        "id": "assignTitle",
+        "isVisible": False,
+    }
+
+    input_card = {
+        "type": "Input.ChoiceSet",
+        "id": "assignInput",
+        "isVisible": False,
+        "choices": teams,
+    }
+
+    submit_card = {
+        "type": "ActionSet",
+        "id": "assignSubmit",
+        "isVisible": False,
+        "actions": [{"type": "Action.Submit", "title": "Assign", "data": {"actionType": "assign"}}],
+    }
+
+    return [title_card, input_card, submit_card]
+
+
+def build_incident_action_cards(group):
+    status = group.get_status()
+    action_cards = []
+    if status != GroupStatus.RESOLVED:
+        action_cards += build_incident_resolve_card()
+    if status != GroupStatus.IGNORED:
+        action_cards += build_incident_ignore_card()
+    action_cards += build_incident_assign_card(group)
+
+    return {"type": "ColumnSet", "columns": [{"type": "Column", "items": action_cards}]}
+
+
+def build_incident_card(group, event, rules):
```

Comment:
@KKaempen The word "incident" is tied to metric alerts, not issue alerts. I think for Slack, it uses the word "group" for issue alerts.

Reply:
Didn't realize that, my bad.